### PR TITLE
Construct email. Fixes #7.

### LINKED ISSUE #7: Mail construction
Construct a mail with a Message, which
- is GPG encrypted
- is GPG signed
- has the given public key attached

Package: mail

### PR DESCRIPTION
**Pull request by JohannesEbke**

### CODE REVIEW

File: gpg/gpg_test.go [hunk 1]
```diff
@@ -166,13 +167,18 @@ func TestGPGCheckMessageSignature(t *testing.T) {
 
 func TestGPGEncryptMessage(t *testing.T) {
 	gpg := setupGPG(t)
+	assert.NotNil(t, gpg)
```
Comment (v-maurer) [inline gpg/gpg_test.go]: This check is not really necessary, since that case would(should) have been caught in setupGPG.

File: gpg/gpg_test.go [hunk 2]
```diff
@@ -166,13 +167,18 @@ func TestGPGCheckMessageSignature(t *testing.T) {
...
 
-	message := bytes.NewReader(testMessageBytes)
 	cipherTextBuffer := new(bytes.Buffer)
 	recipientKeyFile, cleanup := utils.Open(t, asciiKeyFileClient)
 	defer cleanup()
 
-	err := gpg.EncryptMessage(message, cipherTextBuffer, recipientKeyFile)
+	writeCloser, err := gpg.EncryptMessage(cipherTextBuffer, recipientKeyFile)
+	assert.Nil(t, err)
```
Comment (v-maurer) [inline gpg/gpg_test.go]: For errors, you should use assert.NoError as it shows the error message.

File: gpg/gpg_test.go [hunk 3]
```diff
@@ -166,13 +167,18 @@ func TestGPGCheckMessageSignature(t *testing.T) {
...
+	assert.NotNil(t, writeCloser)
+	_, err = writeCloser.Write(testMessageBytes)
+	assert.Nil(t, err)
+	assert.Nil(t, writeCloser.Close())
 	require.NoError(t, err, "Encryption failed")
```
Comment (v-maurer) [inline gpg/gpg_test.go:181]: This check is more less a duplicate of the checks above it right?

File: mail/construct_test.go [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+	"testing"
+
+	"github.com/TNG/gpg-validation-server/gpg"
+	"github.com/TNG/gpg-validation-server/test/utils"
+)
+
+const prefix = "../test/keys/test-gpg-validation@server.local (0x87144E5E) "
+const asciiKeyFilePublic = prefix + "pub.asc"
+const asciiKeyFilePrivate = prefix + "sec.asc"
+const passphrase = "validation"
```
Comment (v-maurer) [inline mail/construct_test.go:15]: Maybe we should move that into a shared package like we did with test/utils.

File: mail/construct_test.go [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+	defer cleanup()
+
+	gpg, err := gpg.NewGPG(serverPrivateKeyFile, passphrase)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	m := OutgoingMail{"It works!", "test-gpg-validation@client.local", clientKey, []byte{}, gpg}
+	b := m.Bytes()
+	log.Print(string(b))
```
Comment (v-maurer) [inline mail/construct_test.go]: There's also testing.T.Log.

File: mail/multipart.go [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+}
+
+func (w *EncodingMultipartWriter) writeFile(name, mimeType, description, disposition string) (io.Writer, error) {
+	err := w.checkWriteHeaders()
+	if err != nil {
+		return nil, err
+	}
+	return w.multipartWriter.CreatePart(textproto.MIMEHeader{
+		"Content-Type":        {mimeType + "; name=\"" + name + "\""},
+		"Content-Disposition": {disposition + "; filename=\"" + name + "\""},
```
Comment (v-maurer) [inline mail/multipart.go:85]: How does this behave if name contains quotation marks?

File: mail/construct_test.go (hunk 4 above)
Comment (JohannesEbke) [inline mail/construct_test.go:15]: next time? :)

File: mail/multipart.go [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+		"Content-Description": {description},
+	})
+}
+
+// WriteInlineFile writes a multipart with an inline file
```
Comment (v-maurer) [inline mail/multipart.go]: It only writes the part header, right? You still have to write the file contents yourself.

**Pull request merged**